File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.py

def solution(n,a,b):
    def matching(a,b):
        if a % 2 == 0:
            return False
        else:
            if a+1 == b:
                return True
            return False
    
    answer = 0

    left = min(a,b)
    right = max(a,b)
    
    while not matching(left,right):
        if left % 2 == 0:
            left = left // 2
        else:
            left = left // 2 + 1
        if right % 2 == 0:
            right = right // 2
        else:
            right = right // 2 + 1
            
        answer+=1
            
    return answer+1

File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.java
import java.util.*;
import java.io.*;

class Solution
{
    public int solution(int n, int a, int b)
    {
        int answer = 1;

        int left = Math.min(a,b);
        int right = Math.max(a,b);
        
        while (!match(left, right)) {
            answer++;
            left = left % 2 == 0? (int)(left / 2) : (int)(left / 2)+1;
            right = right % 2 ==0? (int)(right / 2) : (int)(right / 2)+1;
            
        }

        return answer;
    }
    
    public static boolean match(int a, int b){
        if (a % 2 == 0){
            return false;
        } else{
            if (a+1 == b){
                return true;
            } else{
                return false;
            }
        }
    }
}